Doc/c-api/typeobj.rst

```diff
@@ -664,6 +664,10 @@ and :c:type:`PyType_Type` effectively act as defaults.)
    :c:func:`PyObject_GC_Del` if the instance was allocated using
    :c:func:`PyObject_GC_New` or :c:func:`PyObject_GC_NewVar`.
 
+   Finally, if the type is heap allocated (:const:`Py_TPFLAGS_HEAPTYPE` or
+   built with :c:func:`PyType_FromSpec`), the deallocator should decrement the
+   reference count for its type object.
```

Comment:
Also add that this should specifically happen `after the instance has already been deallocated`. There are two scenarios where this is specially important:
1) The instance was resurrected. In which you should not decref the type.
2) The instance owns the last reference to the type. If you decref the type (which triggers a tp_dealloc on the type itself), now you will have a dangling pointer in the instance's ob_type member which might trigger some segfaults.

Reply:
Done, I also left a suggested implementation like in `tp_finalize` to help explain better (similar to your whatsnew entry). Please look it over